lib/ansible/plugins/action/net_template.py

```diff
@@ -93,6 +93,13 @@ def _handle_template(self):
         except IOError:
             return dict(failed=True, msg='unable to load src file')
 
+        searchpath = [working_path, os.path.dirname(source)]
+        if self._task._role is not None:
+            if self._task._block._dep_chain:
+                for role in sorted(self._task._block.get_dep_chain()):
+                    searchpath.insert(1, role._role_path)
+            searchpath.insert(1, self._task._role._role_path)
```

Comment:
I'm trying to follow the final order of the paths... I think the final order is `[working _path, _role._role_path, *_block._role_paths, dirname(source)]`

Would it not be more straightforward to order it thusly:

    searchpath = [working_path]
    if self._task._role is not None:
        searchpath.append(_role._role_path)
        for role in sorted(dep_chain()):
            searchpath.append(role._role_path)
    searchpath.append(dirname(source))

That would seem to make the intended order clearer while avoiding the overhead of a bunch of insert(1, x) calls